pkg/build/builder/source.go

```diff
@@ -92,6 +97,12 @@ func fetchSource(dockerClient DockerClient, dir string, build *api.Build, urlTim
 		}
 	}
 
+	if len(build.Spec.Source.ContextDir) > 0 {
+		if _, err := os.Stat(filepath.Join(dir, build.Spec.Source.ContextDir)); os.IsNotExist(err) {
+			return sourceInfo, contextDirNotFoundError(filepath.Join(dir, build.Spec.Source.ContextDir))
```

Comment:
just report the contextdir path, "dir" is going to be an internal path with no real meaning to someone who sees it.